apps/api-gateway/src/dtos/authDto.dto.ts

```diff
@@ -1,12 +1,14 @@
 import { IsBoolean, IsEmail, IsNotEmpty, IsOptional, IsString } from 'class-validator';
 
 import { ApiProperty } from '@nestjs/swagger';
+import { Transform } from 'class-transformer';
 
 export class AuthDto {
     @ApiProperty({ example: 'awqx@getnada.com' })
-    @IsEmail()
-    @IsNotEmpty({ message: 'Please provide valid email' })
-    @IsString({ message: 'email should be string' })
+    @IsEmail({}, { message: 'Please provide a valid email' })
+    @IsNotEmpty({ message: 'Email is required' })
+    @IsString({ message: 'Email should be a string' })
+    @Transform(({ value }) =>  'string' === typeof value ? value.trim() : value)
```

Comment:
We have already created a `trim()` function in the `cast.helper.ts` file under library and applied logic of `typeof` in it. use that.

Reply:
Resolved